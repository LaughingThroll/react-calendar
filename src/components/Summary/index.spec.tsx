import { shallow } from 'enzyme'
import SummaryComponent, { Summary } from './'

const setUp = (props: Summary) => shallow(<SummaryComponent {...props} />)

describe('Summary render', () => {
  it('default render', () => {
    const date = new Date('2021-04-21')
    const props = { countUsers: 10, percent: 5, date }
    const tree = setUp(props)

    expect(tree).toMatchSnapshot()
    expect(tree.text().includes('April')).toBe(true)
  })

  it('render with other props', () => {
    const date = new Date('2021-03-21')
    const props = { countUsers: -10, percent: -5, date }
    const tree = setUp(props)

    expect(tree.text().includes('March')).toBe(true)
  })
})
