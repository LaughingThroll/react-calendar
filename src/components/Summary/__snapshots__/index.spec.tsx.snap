// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary render default render 1`] = `
<div
  className="summary"
>
  <div
    className="summary__title"
  >
    April
     teams Summary
  </div>
  <div
    className="summary__inner"
  >
    <div
      className="summary__text"
    >
      On vacation
    </div>
    <Memo(CountUsers)
      classNames={
        Array [
          "summary__users",
        ]
      }
      countUsers={10}
    />
    <Memo(WeekendPercents)
      classNames={
        Array [
          "summary__weekendPercent",
        ]
      }
      percent={5}
    />
  </div>
</div>
`;
